### ISSUE TITLE
can`t run rancher agent on docker with storagedriver is overlay

### ISSUE THREAD

**Issue by yutian1224** [rancher/rancher#3652]
Version: docker-1.9.1    agent-0.9.0
[root@test482 docker]# docker info
Containers: 0
Images: 18
Server Version: 1.9.1
Storage Driver: overlay
 Backing Filesystem: xfs
Execution Driver: native-0.2
Logging Driver: json-file
Kernel Version: 3.10.0-229.el7.x86_64
Operating System: Red Hat Enterprise Linux Server 7.1 (Maipo)
CPUs: 24
Total Memory: 62.73 GiB
Name: test482
ID: HKUW:HMQV:AOG3:MBG2:2KXQ:HRDI:POJ5:BKAP:46G3:I3MI:YNHZ:H3MK

Steps: 
run rancher agent on docker server
`sudo docker run -d --privileged -v /var/run/docker.sock:/var/run/docker.sock -v /var/lib/rancher:/var/lib/rancher stargate.lufax.tool/rancher/agent:v0.9.0 http://172.19.48.3:8080/v1/scripts/D8411D6DCA17E45CE30E:1455804000000:4zxJTziV47nQKQC0xrWqySi1szw`

Results: 
`[root@test482 docker]# sudo docker run -d --privileged -v /var/run/docker.sock:/var/run/docker.sock -v /var/lib/rancher:/var/lib/rancher stargate.lufax.tool/rancher/agent:v0.9.0 http://172.19.48.3:8080/v1/scripts/D8411D6DCA17E45CE30E:1455804000000:4zxJTziV47nQKQC0xrWqySi1szw
Error response from daemon: mkdir /data/docker/overlay/ea1047215e2fa4578ddbe04007db2372e3df69bea2efc55cfae9f5b24ff1a93a-init/merged/dev/shm: invalid argument`

Expected:
docker config
`INSECURE_REGISTRY="--insecure-registry=stargate.lufax.tool"
OPTIONS="-s overlay"
DOCKER_NETWORK_OPTIONS="-H tcp://0.0.0.0:2375 -H unix:///var/run/docker.sock"
DOCKER_STORAGE_OPTIONS="-g=/wls/docker"`

**Comment by yutian1224** [comment]
Did any people have the same problem ?

**Comment by deniseschannon** [comment]
What Rancher server version are you using? 

What is the rancher/agent version you are using? It doesn't look like the default one provided by Rancher and we've never released v0.9.0 with any release.

**Comment by yutian1224** [comment]
Rancher server : v0.59.0
My fault.I update rancher/agent to default version for rancher/server(v0.9.2).But it`s still not working.
```
[root@test481 ~]# sudo docker run -d --privileged -v /var/run/docker.sock:/var/run/docker.sock -v /var/lib/rancher:/var/lib/rancher stargate.lufax.tool/rancher/agent:v0.9.2 http://172.19.48.3:8080/v1/scripts/D8411D6DCA17E45CE30E:1456394400000:KKNGWKWXorWIfYBRiGb2AKlCmA
docker: Error response from daemon: mkdir /data/docker/overlay/fa2810f5c62cdfc174482b8f23142f9b210169b1b6e97f322521a17a2e3efbb8-init/merged/dev/shm: invalid argument.
See 'docker run --help'.
```

**Comment by yutian1224** [comment]
I realised that docker can not work prefectly on RHEL/Centos.
The problem is gone when I change node to Ubuntu.

**Comment by deniseschannon** [comment]
What version of RHEL/CentOS were you trying?

**Comment by yutian1224** [comment]
Operating System: Red Hat Enterprise Linux Server 7.1 (Maipo)

**Comment by ibuildthecloud** [comment]
I believe overlay on XFS does not work well.  We have seen overlay on ext4 work on RHEL 7.2.

**Comment by VIPv6** [comment]
I've experience the same problem today.  I'd be happy to change to Ubuntu, but we've invested a fair amount into Centos 7.2.  This is NOT using XFS.  I've done minimal security measures.  No SELinux yet. I've secured the SSH logins a bit and we're using SSH keys:

Unable to find image 'rancher/server:latest' locally
Trying to pull repository docker.io/rancher/server ... 
latest: Pulling from docker.io/rancher/server
96c6a1f3c3b0: Pull complete 
ed40d4bcb313: Pull complete 
b171f9dbc13b: Pull complete 
ccfc4df4fbba: Pull complete 
9e4d5a1238bb: Pull complete 
f986ba0d224d: Pull complete 
303d9649bb89: Pull complete 
56825bfb5bb7: Pull complete 
86a35696481e: Pull complete 
eb2848176ff4: Pull complete 
fabbcc718243: Pull complete 
35489637da84: Pull complete 
5c005e2df0e8: Pull complete 
64ba16c4c9f4: Pull complete 
Digest: sha256:d5a798d1274bcf6813fc9866660dc8559b7e17cdce47608bce28d134bd4f2dc1
Status: Downloaded newer image for docker.io/rancher/server:latest
docker: Error response from daemon: mkdir /var/lib/docker/overlay/dcf7e6e7e7c1dcde047262af175b0f2d835a9dbe7e9c68daf44fb5fb87430af4-init/merged/dev/shm: invalid argument.
See '/usr/bin/docker-current run --help'.
